### PR TITLE
feat(installation-spell): return logs as result type [NET-592]

### PR DESCRIPTION
**Pull request by justprosh**

### CODE REVIEW

**Review by kmd-fl — APPROVED** [review]

File: fluence.yaml [hunk 1]
```diff
@@ -29,11 +29,12 @@
 # appTSPath: ./path/to/app-ts/dir # Optional. If you want to generate ts files including app.ts to be able to access deployed app data in aqua when using FluenceJS,
 # appJSPath: ./path/to/app-js/dir # Optional. If you want to generate js files including app.js to be able to access deployed app data in aqua when using FluenceJS
 
-version: 2
+version: 3
 
 dependencies:
   npm:
     "@fluencelabs/cli": 0.5.4
```
Comment (folex) [inline fluence.yaml:36]: why do we have it here still?

File: src/aqua/installation-spell/src/aqua/cli.aqua [hunk 2]
```diff
@@ -13,52 +13,66 @@ func upload_deploy(config: LocalAppDeployConfig) -> DeployedAppWorkers:
...
 
     on HOST_PEER_ID:
         for deal_id <- deals:
-            result <- Subnet.resolve(deal_id)
-            if result.success == false:
+            subnet_res <- Subnet.resolve(deal_id)
+            if !subnet_res.success:
                 on INIT_PEER_ID via HOST_PEER_ID:
-                    co Console.print(["Failed to resolve subnet for deal_id: ", deal_id, " error: ", result.error])
+                    result <<- LogsDealResult(deal_id = deal_id, success = false, error = subnet_res.error, logs = nil)
```
Comment (folex) [inline src/aqua/installation-spell/src/aqua/cli.aqua:58]: what will be in the error message here?

File: src/aqua/installation-spell/src/aqua/cli.aqua [hunk 3]
```diff
@@ -13,52 +13,66 @@ func upload_deploy(config: LocalAppDeployConfig) -> DeployedAppWorkers:
...
-                            co Console.print(["Worker for deal_id:", deal_id, "and pat_id", w.pat_id, "is not installed yet"])
+                            -- worker is not installed yet
+                            logs <<- LogsDeal(success = true, error = nil, spell_id = nil, worker_id = nil, host_id = host_id, logs = nil)
                     else:
                         on worker_id! via host_id:
                             spell_id <- Srv.resolve_alias("worker-spell")
                             Spell spell_id
                             get_res <- Spell.get_logs()
-                            logs <<- LogDeal(spell_id = spell_id, worker_id = worker_id!, host_id = host_id, deal_id = deal_id, logs = get_res.logs)
+                            logs <<- LogsDeal(success = get_res.success, error = ?[get_res.error], spell_id = ?[spell_id], worker_id = worker_id, host_id = host_id, logs = get_res.logs)
```
Comment (folex) [inline src/aqua/installation-spell/src/aqua/cli.aqua:74]: what text will be in the error here?

**Review by folex — APPROVED** [review]

File: src/aqua/installation-spell/src/aqua/cli.aqua [hunk 4]
```diff
@@ -13,52 +13,66 @@ func upload_deploy(config: LocalAppDeployConfig) -> DeployedAppWorkers:
...
+                logs <<- LogsDirectResult(success = get_res.success, error = ?[get_res.error], spell_id = i.spell_id, worker_id = i.worker_id, host_id = i.host_id, logs = get_res.logs, worker_name = w.name)
 
     <- logs
 
-data LogDeal:
-    spell_id: string
-    worker_id: string
+data LogsDeal:
+    success: bool
+    error: ?string
```
Comment (shamsartem) [inline src/aqua/installation-spell/src/aqua/cli.aqua:38]: is it possible to not have separate success and error and just have optional error. If error doesn't exist - then it's a success?

File: src/aqua/installation-spell/src/aqua/cli.aqua (hunk 2 above)
Comment (shamsartem) [reply src/aqua/installation-spell/src/aqua/cli.aqua:58]: I am also curious how should I display it. We have nested kind of errors here.
In the test should I check for existance of my error message?
e.g. if I display this like that:

`Couldn't resolve subnet. Error: <subnet_res.error>`

in tests I would check for `Couldn't resolve subnet. Error:` string and if exists I will fail test. Sounds ok?

File: fluence.yaml (hunk 1 above)
Comment (justprosh) [reply fluence.yaml:36]: You mean .yaml? I think I can remove it and put in gitignore

File: src/aqua/installation-spell/src/aqua/cli.aqua (hunk 4 above)
Comment (justprosh) [reply src/aqua/installation-spell/src/aqua/cli.aqua:38]: yeah, i think I can do that

File: src/aqua/installation-spell/src/aqua/cli.aqua (hunk 2 above)
Comment (justprosh) [reply src/aqua/installation-spell/src/aqua/cli.aqua]: Now it looks like: `Failed to resolve subnet: .....`

File: src/aqua/installation-spell/src/aqua/cli.aqua (hunk 3 above)
Comment (justprosh) [reply src/aqua/installation-spell/src/aqua/cli.aqua]: `Failed to get logs: ....`

**Review by folex — APPROVED** [review]

**Pull request merged**